doc/source/user/tutorial-ma.rst

```diff
@@ -0,0 +1,377 @@
+comes with a specific implementation of most :term:`NumPy universal functions
+(ufuncs) <ufunc>`, which means that you can still apply fast vectorized
+functions and operations on masked data. The output is then a masked array.
+We'll see some examples of how this works in practice below.
+
+**Using masked arrays to see COVID-19 data**
+
+From `Kaggle <https://www.kaggle.com/atilamadai/covid19>`_ it is possible to
+download a dataset with initial data about the COVID-19 outbreak in the
+beginning of 2020. We are going to look at a small subset of this data,
+contained in the file ``who_covid_19_sit_rep_time_series.csv``.
+
+.. ipython:: python
+
+   import numpy as np
+   import os
+   # The os.getcwd() function returns the current folder; you can change
+   # the filepath variable to point to the folder where you saved the .csv file.
+   filepath = os.getcwd()
+   @suppress
+   filepath = os.path.join(filepath, "source", "user")
+   filename = os.path.join(filepath, "who_covid_19_sit_rep_time_series.csv")
+   
+The data file contains data of different types and is organized as follows:
+
+- The first row is a header line that (mostly) describes the data in each column
+  that follow in the rows below, and beginning in the fourth column, the header
+  is the date of the observation.
+- The second through seventh row contain summary data that is of a different
+  type than that which we are going to examine, so we will need to exclude that
+  from the data with which we will work.
+- The numerical data we wish to work with begins at column 4, row 8, and extends
+  from there to the rightmost column and the lowermost row.
+
+Let's explore the data inside this file for the first 14 days of records. To
+gather data from the ``.csv`` file, we will use the :func:`numpy.genfromtxt`
+function, making sure we select only the columns with actual numbers instead of
+the first three columns which contain location data. We also skip the first 7
+rows of this file, since they contain other data we are not interested in.
+Separately, we will extract the information about dates and location for this
+data. 
+
+.. ipython:: python
+
+   # Note we are using skip_header and usecols to read only portions of the data
+   # file into each variable.
+   # Read just the dates for columns 3-7 from the first row
+   dates = np.genfromtxt(filename, dtype=np.unicode_, delimiter=",",
+	                 max_rows=1, usecols=range(3, 17),
+			 encoding="utf-8-sig")
+   # Read the names text names of the geographic locations from the first two
+   # columns, skipping the first seven rows
+   locations = np.genfromtxt(filename, dtype=np.unicode_, delimiter=",",
+	                     skip_header=7, usecols=(0, 1),
+			     encoding="utf-8-sig")
+   # Read the numeric data from just the first 14 days
+   nbcases = np.genfromtxt(filename, dtype=np.int_, delimiter=",",
+                           skip_header=7, usecols=range(3, 17),
+			   encoding="utf-8-sig")
```

Comment:
Indents are all over the place here - part of the problem is you've indented the code block 3 spaces and not 4, the other problem is you're mixing tabs and spaces. I'd recommend either:
```suggestion
    dates = np.genfromtxt(
        filename, dtype=np.unicode_, delimiter=",", max_rows=1,
        usecols=range(3, 17), encoding="utf-8-sig")
    # Read the names text names of the geographic locations from the first two
    # columns, skipping the first seven rows
    locations = np.genfromtxt(
        filename, dtype=np.unicode_, delimiter=",", skip_header=7,
        usecols=(0, 1), encoding="utf-8-sig")
    # Read the numeric data from just the first 14 days
    nbcases = np.genfromtxt(
        filename, dtype=np.int_, delimiter=",", skip_header=7,
        usecols=range(3, 17), encoding="utf-8-sig")
```
or
```suggestion
    dates = np.genfromtxt(filename, dtype=np.unicode_, delimiter=",",
                          max_rows=1, usecols=range(3, 17),
                          encoding="utf-8-sig")
    # Read the names text names of the geographic locations from the first two
    # columns, skipping the first seven rows
    locations = np.genfromtxt(filename, dtype=np.unicode_, delimiter=",",
                              skip_header=7, usecols=(0, 1),
                              encoding="utf-8-sig")
    # Read the numeric data from just the first 14 days
    nbcases = np.genfromtxt(filename, dtype=np.int_, delimiter=",",
                            skip_header=7, usecols=range(3, 17),
                            encoding="utf-8-sig")
```

Reply:
I think this was a copy-paste problem, and tabs got inserted somehow. Sorry!